main.ts

```diff
@@ -109,12 +110,33 @@ export default class SmarterMDhotkeys extends Plugin {
 				// https://github.com/obsidianmd/obsidian-api/blob/fac5e67f5d83829a4e0126905494c8cbca27765b/obsidian.d.ts#L787
 
 				if (editor.cm instanceof window.CodeMirror) return editor.cm.findWordAt(ep); // CM5
+				
+				// get text around cursor
+				const currentPosition = editor.posToOffset(ep);
+				const textBeforeCursor = editor.cm.state.sliceDoc(currentPosition - 1000, currentPosition);
+				const textAfterCursor = editor.cm.state.sliceDoc(currentPosition, currentPosition + 1000);
+
+				const inTag = textBeforeCursor.match(/#[\w\/\_\-]*$/)

```

Comment:
That won't match tags that contain non-ASCII characters.

Reply:
you have a regex that properly matches tags with non-ASCII characters?

Reply:
https://github.com/slevithan/xregexp/issues/228#issuecomment-811478498

Maybe this helps? Otherwise I'd use a regex that matches everything from `#` until the first `\s` and test whether the part after `#` is a number. If it is, discard it, otherwise keep it.